CMakeLists.txt

```diff
@@ -58,7 +58,7 @@ option(PRECICE_PETScMapping "Enable use of the PETSc linear algebra library." ON
 option(PRECICE_PythonActions "Python support" ON)
 option(PRECICE_Packages "Configure package generation." ON)
 option(PRECICE_InstallTest "Add test binary and necessary files to install target." OFF)
-option(BUILD_SHARED_LIBS "Build shared libraries by default" OFF)
+option(BUILD_SHARED_LIBS "Build shared libraries by default" ON)
```

Comment:
I like the change. However, the `CMake` description is a bit weird as the last argument denotes just the default argument. Launching the cmake-gui renders just `Build shared libraries by default` as argument description, which is confusing. Maybe something like `Build preCICE as a shared library (default is ON)` would fit better if you want to inform the user about the defualt configuration.

Reply:
I agree that this looks a bit odd.

What about simply `Build shared instead of static libraries`.
The default is what the user sees first, so it shouldn't be necessary to explicitly print it. This would also make it inconsistent with the other options.

Reply:
Sounds good :+1: